Sources/TuistKit/Services/InitService.swift

```diff
@@ -72,16 +66,29 @@ class InitService {
     ) {
         let path = self.path(path)
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
+        var attributes: [Template.Attribute] = []
 
-        let templateDirectory = try templateDirectory(
-            templateDirectories: directories,
-            template: templateName
-        )
+        if templateName.isGitURL {
+            try templateGitLoader.loadTemplate(from: templateName) { template in
+                attributes = template.attributes
+            }
+        } else {
+            let templateDirectory = try templateDirectory(
+                templateDirectories: directories,
+                template: templateName
+            )
 
-        let template = try templateLoader.loadTemplate(at: templateDirectory)
+            let template = try templateLoader.loadTemplate(at: templateDirectory)
+            attributes = template.attributes
+        }
 
-        return template.attributes
+        return attributes
             .reduce(into: (required: [], optional: [])) { currentValue, attribute in
+                // name and platform attributes have default values, so add them to the optional
+                if attribute.name == "name" || attribute.name == "platform" {
+                    currentValue.optional.append(attribute.name)
+                    return
+                }
```

Comment:
Let's assume that the required `name` attribute is set in the template. Then, if you call `tuist init` without specifying this attribute, an error will occur:

```
Error: Missing expected argument '--name <name>'
Help:  --name <name>  
Usage: tuist init [--platform <platform>] [--template <template>] [--path <path>] --name <name> --test <test>
  See 'tuist init --help' for more information.
``` 

At the same time:

```
tuist init --help
OVERVIEW: Bootstraps a project

USAGE: tuist init [--platform <platform>] [--name <name>] [--template <template>] [--path <path>]

OPTIONS:
  --platform <platform>   The platform (ios, tvos or macos) the product will be
                          for (Default: ios)
  -n, --name <name>       The name of the project. If it's not passed (Default:
                          Name of the directory)
  -t, --template <template>
                          The name of the template to use (you can list
                          available templates with tuist scaffold list)
  -p, --path <path>       The path to the folder where the project will be
                          generated (Default: Current directory)
  -h, --help              Show help information.
```

In other words, if you do not specify `name` and `platform` attributes, then they take on the default values. Therefore, if these attributes were found in the template, I forced them into the list of optionals so that they were assigned default values.